### PR TITLE
Added new optional interface to safe_convert

### PR DESCRIPTION
**Pull request by kfeinauer**

### CODE REVIEW

File: src/cpp/core/include/core/SafeConvert.hpp [hunk 1]
```diff
@@ -23,6 +23,7 @@
 
 #include <boost/lexical_cast.hpp>
 #include <boost/numeric/conversion/cast.hpp>
+#include <boost/optional.hpp>
```
Comment (gtritchie) [inline src/cpp/core/include/core/SafeConvert.hpp]: Can we just use <optional> here? Now that we're c++11 should prefer standard library stuff over boost.

Comment (kfeinauer) [reply src/cpp/core/include/core/SafeConvert.hpp]: I agree in principal, but we already use boost::optional in a lot of places and the interface is not the same.

Comment (gtritchie) [reply src/cpp/core/include/core/SafeConvert.hpp]: Oh, and <optional> wasn't added to the standard until c++17, if I'm reading the Internet correctly. Prior to that it was, ahem, optional (technical spec). Thus I should probably go back and remove my usages of <optional> (though we haven't hit an issue yet, so perhaps very commonly implemented in c++ 11.

Comment (gtritchie) [reply src/cpp/core/include/core/SafeConvert.hpp]: I thought it was C++11.

Comment (gtritchie) [reply src/cpp/core/include/core/SafeConvert.hpp]: Otherwise, LGTM!

**Comment by jmcphers** [comment]
Very nice, I think this will get a lot of usage!

**Pull request merged**